cccatalog-api/cccatalog/api/views/image_views.py

```diff
@@ -58,28 +59,89 @@ def _get_user_ip(request):
 
 
 class SearchImages(APIView):
-    """
-    Search for images by a query string. Optionally, filter results by specific
-    licenses, or license "types" (commercial use allowed, modification allowed,
-    etc). Results are ranked in order of relevance.
-
-    Refer to the Lucene syntax guide for information on structuring advanced
-    searches. https://lucene.apache.org/core/2_9_4/queryparsersyntax.html
-
-    Although there may be millions of relevant records, only the most relevant
-    several thousand records can be viewed. This is by design: the search
-    endpoint should be used to find the top N most relevant results, not for
-    exhaustive search or bulk download of every barely relevant result.
-    As such, the caller should not try to access pages beyond `page_count`,
-    or else the server will reject the query.
-    """
+    image_search_description = \
+        """
+        Search for images by a query string. Optionally, filter results by specific
+        licenses, or license "types" (commercial use allowed, modification allowed,
+        etc). Results are ranked in order of relevance.
+
+        Refer to the Lucene syntax guide for information on structuring advanced
+        searches. https://lucene.apache.org/core/2_9_4/queryparsersyntax.html
+
+        Although there may be millions of relevant records, only the most relevant
+        several thousand records can be viewed. This is by design: the search
+        endpoint should be used to find the top N most relevant results, not for
+        exhaustive search or bulk download of every barely relevant result.
+        As such, the caller should not try to access pages beyond `page_count`,
+        or else the server will reject the query.
+
+        Example using single query parameter:
+    
+        ```
+        $ curl -H "Authorization: Bearer DLBYIcfnKfolaXKcmMC8RIDCavc2hW" https://api.creativecommons.engineering/v1/images?q=test
+        ```
+
+
+        Example using multiple query parameters:
+
+        ```
+        $ curl -H "Authorization: Bearer DLBYIcfnKfolaXKcmMC8RIDCavc2hW" https://api.creativecommons.engineering/v1/images?q=test&license=pdm,by&categories=illustration&page_size=1&page=1
+        ```
+
+        """  # noqa
+    image_search_response = {
+        "200": openapi.Response(
+            description="OK",
+            examples={
+                "application/json": {
+                    "result_count": 77,
+                    "page_count": 77,
+                    "page_size": 1,
+                    "results": [
+                        {
+                            "title": "File:Well test separator.svg",
+                            "id": "36537842-b067-4ca0-ad67-e00ff2e06b2d",
+                            "creator": "en:User:Oil&GasIndustry",
+                            "creator_url": "https://en.wikipedia.org/wiki/User:Oil%26GasIndustry",  # noqa
```

Comment:
As you've discovered, there's a style test requirement that lines don't exceed a certain length. This helps us keep our Python code simpler, but it gets in the way when we're trying to show big JSON responses. Instead of using `# noqa`  silence each individual line that exceeds the limit, it would be better to use a broader exemption for the entire example response.

I think you can achieve this by adding `# noqa` after the closing brace for the image_search_response definition.

```
image_search_response = {
    . . .
} # noqa: E501
```

(`E501` is the [name of the line length violation](https://pycodestyle.pycqa.org/en/latest/intro.html#error-codes) in the PEP 8 spec; it's best practice to only break one rule at a time)

Reply:
I added `# noqa: E501` after the closing brace for the image_search_response definition but I got errors in the style test.

![image](https://user-images.githubusercontent.com/10581923/95604660-7df67300-0a8a-11eb-88c0-0517494871ec.png)

Reply:
That's unfortunate, it seems like pycodestyle only allows you to ignore individual strings and lines. I did some research and it looks like the only option is to exclude the file from pycodestyle entirely, which we wouldn't really want to do to something as central as the view code.

Given those constraints, here's what I recommend:

1. Create a separate file for your JSON response examples in `cccatalog-api/cccatalog/example_responses.py`
2. Configure `pycodestyle` to ignore this file. See this line:       run: pycodestyle cccatalog-api/cccatalog --exclude='cccatalog-api/cccatalog/api/migrations' --max-line-length=80 --ignore=E402,E702
3. Import examples in each view.

That way, we can ignore style violations in the JSON examples specifically without impacting other parts of the code.

Reply:
If I were to follow your recommendation, I'd have to import class `AboutImageResponse` into `cccatalog-api/cccatalog/example_responses.py`. Isn't cyclic dependency a bad practice?

https://github.com/creativecommons/cccatalog-api/blob/e976f93207e0d4fabd24dd546fd06601484cda1c/cccatalog-api/cccatalog/api/views/site_views.py#L50-L87

Also, I figured out that I can just separate the value into multiple lines.

https://github.com/creativecommons/cccatalog-api/blob/e976f93207e0d4fabd24dd546fd06601484cda1c/cccatalog-api/cccatalog/api/views/site_views.py#L171-L186

Reply:
Yes, cyclic imports will typically mess things up in Python and cause a number of headaches. However, I think you can avoid importing anything into `example_responses.py` if you exclusively store the contents of the `openapi.Response(examples={...})` parameter and nothing else.

Line breaking the dict values does follow the letter of pep8 but not the spirit; I find it harder to read the examples inline after making that change

Reply:
Alright, updated.